impeller/entity/entity_pass_clip_stack.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/entity/entity_pass_clip_stack.h"
+#include "impeller/entity/contents/clip_contents.h"
+#include "impeller/entity/contents/content_context.h"
+#include "impeller/entity/entity.h"
+
+namespace impeller {
+
+EntityPassClipStack::EntityPassClipStack() {}
+
+std::optional<Rect> EntityPassClipStack::CurrentClipCoverage() const {
+  return clip_coverage_.back().back().coverage;
+}
+
+bool EntityPassClipStack::HasCoverage() const {
+  return !clip_coverage_.back().empty();
+}
+
+void EntityPassClipStack::Initialize(const Rect& rect) {
```

Comment:
nit: It would be nice if this was done in the constructor instead, but I am also not sure how feasible that is

Reply:
Done